### ISSUE TITLE
Make a system specific /system directory build

### ISSUE THREAD

**Issue by R0NAM1** [AsteroidOS/meta-smartwatch#38]
Right now Skipjack uses a generic System Directory, while this seems to be working fine for now, eventually I would like to make it system specific, especially as I already have the stock partition dumps.

**Comment by neilgfoster** [comment]
Some evidence from a TicWatch C2+ (`skipjack`) suggesting this is worth keeping open, with a specific remaining gap.

Up front: I did this investigation with an AI assistant (Claude Code) driving the analysis, on my own hardware. Every command was run against this physical watch and every quoted output below is real, not generated — but the interpretation had AI involvement, including two wrong conclusions I retract further down. Please read it with the scepticism that warrants, and tell me if you'd rather not have AI-assisted reports on the tracker.

**The shipped tarball's base is still sawfish.** I RAM-booted AsteroidOS 2.1 for skipjack (`fastboot boot`, rootfs on `/sdcard` — nothing written to flash) and read the `/system` it ships:

```
ro.product.model     = TicWatch C2+
ro.product.name      = skipjack
ro.product.device    = skipjack
ro.build.fingerprint = huawei/sawfish/sawfish:9/PWH1.180422.008/4758795:user/release-keys
```

`name`/`device` are overridden to `skipjack`, but the underlying build is sawfish (Huawei Watch 2). That matches what @R0NAM1 describes above — "originally for sawfish, but with skipjack stuff added / replacing sawfish stuff" — so this looks like the iterative overlay, not the from-scratch pull `README-system-dir` describes. (The tarball currently referenced by `android-system-data_skipjack-p.bb`, sha256 `3e854b7c…`, isn't either of the hashes posted in this thread, so it's a later build than those.)

The overlay approach worked for the things that got overlaid — bluetooth here, sensors in #39, wifi in #42. Three areas appear never to have been covered, and they're exactly the three features that don't work on my unit:

| Function | Shipped (sawfish) | skipjack needs | Result |
|---|---|---|---|
| NFC | `nfc_nci.pn54x.default.so` (NXP PN54x) | `nfc_nci.st21.default.so` (ST ST21NFC) | Wrong chip entirely — NFC cannot work |
| Audio | `audio.primary.sawfish.so` | `audio.primary.msm8909.so` | Wrong tuning; mic routing doesn't match |
| GPS | GNSS shim (`android.hardware.gnss@1.0-impl.so`, `libizat_core.so`) present, but **no `gps.msm8909.so`** | `gps.msm8909.so` + `libloc_*` | Shim has no module to load |

So the three open feature gaps on skipjack look like one cause: they're the parts of the sawfish base that the overlay never replaced.

Two corrections I need to make, because I asserted both and both were wrong — they change how the GPS gap looks:

- **The modem does come up under AsteroidOS.** `pil-q6v5-mss` loads it, `apr_tal: Modem Is Up`, and `/sys/bus/msm_subsys/devices/subsys2` reports `modem: ONLINE`. My earlier "it never comes up" reading came from a dmesg captured late in a long session, after the ring buffer had wrapped — absence of log lines, not absence of the event.
- **`rmt_storage` runs**, registering `modemst1/2`, `fsc` and `fsg` exactly as stock does, and `qmuxd`/`netmgrd` are present. GPS was never blocked on the modem or a missing client.

The hardware side is fine: `/dev/st21nfc` exists under AsteroidOS with the DT node correct (I2C `0x08`, reset GPIO 33, IRQ GPIO 35), and stock Wear OS gets a GPS lock in ~3 s on the same unit. This looks userspace-only. Worth noting the NFC target here is HCE — tag reading doesn't appear to exist on this hardware at all.

I have verified `/system` and `/vendor` captured from this unit before anything was written to flash (raw partition images plus extracted trees, and device trees), so I have the input for the overlay.

I'm happy to do the work and open a PR extending the overlay with those three, following the #39/#42 pattern — that PR would also be AI-assisted, same caveat as above. Before I start, two questions:

1. Is extending the existing overlay the direction you want, or would you rather someone finally did the clean from-scratch `README-system-dir` pull? The latter is more work but ends the sawfish-base problem permanently. I'd rather build the one you'll take.
2. `README-system-dir` says to pull from WearOS 8.0.0, but the recipe is `PV = "pie"` and my unit ships Wear OS 9. If the clean rebuild is the preferred route, which Android base should it target? I couldn't find an Android 9 branch in `AsteroidOS/android_manifest`, which seems like it'd block the bionic/init part of those instructions.

Separately, and not blocking any of the above: `android-system-data_skipjack-p.bb` fetches from a single personal domain (`mosushi.de`) with no mirror. The sha256 is pinned so integrity is covered, but availability isn't — if that host goes away, skipjack stops building. Happy to raise that as its own issue if it isn't already tracked.
